linkerd/stack/src/watch.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{NewService, Param, Service};
+use std::task::{Context, Poll};
+use tokio::sync::watch;
+use tracing::Instrument;
+
+pub trait UpdateWatch<T>: Send + Sync + 'static {
+    type Service: Clone + Default + Send + Sync + 'static;
+
+    fn update(&mut self, target: &T) -> Option<Self::Service>;
+}
+
+#[derive(Clone, Debug)]
```

Comment:
this will only implement `Clone`/`Debug` if `P` is `Clone`/`Debug`, because of the `PhantomData`...should we have a manual impl?

Reply:
yeah probably